### PR TITLE
fix: return non-zero exit code when status unavailable

### LINKED ISSUE #22466: `minikube status` exits with 0 when there is no minikube container
### What Happened?

Running `minikube status` on a system without a minikube container we get the following error:

```
$ minikube status
E0116 14:50:53.995077 3483121 status.go:179] status error: host: state: unknown state "minikube": docker container inspect minikube --format={{.State.Status}}: exit status 1
stdout:


stderr:
Error response from daemon: No such container: minikube
E0116 14:50:53.995092 3483121 status.go:126] status error: host: state: unknown state "minikube": docker container inspect minikube --format={{.State.Status}}: exit status 1
stdout:


stderr:
Error response from daemon: No such container: minikub
``` 

... and what's really bad is that this exits with `0` (success). This used to work find with earlier versions of minikube.

The impact of this is that it causes major issues in my pipelines as I use `if minikube status &>/dev/null; then ... fi;` to detect if the user already has a minikube cluster running.

## reproduce

1) `minikube stop`
2) `docker system prune -a`
3) `minikube status; echo $?`

### minikube version

```
$ minikube version
minikube version: v1.37.0
commit: v1.37.0
```

### Attach the log file

```
$ minikube status
E0116 14:50:53.995077 3483121 status.go:179] status error: host: state: unknown state "minikube": docker container inspect minikube --format={{.State.Status}}: exit status 1
stdout:


stderr:
Error response from daemon: No such container: minikube
E0116 14:50:53.995092 3483121 status.go:126] status error: host: state: unknown state "minikube": docker container inspect minikube --format={{.State.Status}}: exit status 1
stdout:


stderr:
Error response from daemon: No such container: minikub
``` 

### Operating System

Redhat/Fedora

### Driver

Docker

### PR DESCRIPTION
**Pull request by thc1006**
**What this PR does**

Fixes `minikube status` returning exit code 0 when the Docker container does not exist.

**Root cause**: When `cluster.GetStatus()` fails, it returns `nil` statuses. The `exitCode()` function's for-loop doesn't execute for nil/empty slices, returning 0.

**Fix**: Add nil/empty check at the start of `exitCode()` to return 7 (all components not running).

**Changes**
- `cmd/minikube/cmd/status.go`: Add nil check in `exitCode()` (+3 lines)
- `cmd/minikube/cmd/status_test.go`: Add test cases for nil and empty statuses (+16 lines)

**Verification**
- `exitCode()` function: 100% test coverage
- All unit tests pass
- Lint passes (0 issues)
- Manual test: exit code = 7 when container missing

Fixes #22466

### CODE REVIEW

File: cmd/minikube/cmd/status_test.go [hunk 1]
```diff
@@ -43,6 +43,22 @@ func TestExitCode(t *testing.T) {
...
+
+	// 當 GetStatus 失敗時，statuses 會是 nil，應該回傳非零的退出碼
+	t.Run("nil", func(t *testing.T) {
+		got := exitCode(nil)
+		if got != 7 {
+			t.Errorf("exitCode(nil) = %d, want: 7", got)
+		}
+	})
+
+	// 空的 slice 也一樣要回傳非零
```
[the remark this answers is not in the document]
Comment (thc1006) [reply cmd/minikube/cmd/status_test.go]: Sor I should use English

**Comment by thc1006** [comment]
## CI Failure Analysis

Hi maintainers,

I have investigated the CI failures and confirmed they are unrelated to this PR's changes. Here is my analysis:

### Summary of Changes

This PR only modifies the `exitCode()` function in `status.go` (3 lines) and adds corresponding unit tests (16 lines). The `exitCode()` function now has 100% test coverage.

### CI Failures Are Pre-existing Issues

I verified that the exact same tests are failing on recently merged PRs:

**PR #22484** (merged yesterday) has the same failures:
- `integration-docker-containerd-linux-arm`
- `integration-docker-containerd-linux-x86`
- `integration-docker-crio-linux-x86`
- `integration-vfkit-docker-macos-arm`
- `unit_test (windows-2022)`

**PR #22481** (merged two days ago) also has these failures plus additional ones.

### Root Causes (Not Related to This PR)

1. **Windows Unit Tests**: These are pre-existing flaky tests with path format issues (`\` vs `/`) and shell detection problems. For example, `TestMigrateHostV1ToHostV2` expects Windows path format but receives Unix format.

2. **Integration Tests (containerd/crio)**: The `TestDockerEnvContainerd` test has a bug where it uses `t.Errorf` instead of `t.Fatalf`, causing subsequent code to panic with `index out of range [1] with length 0` when the regex match fails.

### Verification

I confirmed that:
- No integration tests depend on empty statuses returning exit code 0
- All unit tests pass locally
- The change is minimal and focused on the reported issue #22466

Please consider running `/retest` to verify. Thank you for your review!
